### PR TITLE
bigquery-emulator 0.6.1 (new formula)

### PR DESCRIPTION
**Pull request by gmpify**
This commit adds a new formula for repo https://github.com/goccy/bigquery-emulator The binary installed after this can be used to emulate a GCP BigQuery server on your local machine

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/b/bigquery-emulator.rb
Comment (chenrui333) [inline Formula/b/bigquery-emulator.rb:23]: We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

File: Formula/b/bigquery-emulator.rb
Comment (chenrui333) [inline Formula/b/bigquery-emulator.rb:6]: try to use github tarball build

File: Formula/b/bigquery-emulator.rb
Comment (gmpify) [reply Formula/b/bigquery-emulator.rb]: Hi @chenrui333, thanks for the speedy review. I've added better test and reviewed the Linux installation process.

File: Formula/b/bigquery-emulator.rb
Comment (chenrui333) [inline Formula/b/bigquery-emulator.rb:33]: we can just drop the `--project` flag test

```suggestion
    port = free_port

    fork do
      exec bin/"bigquery-emulator", "--project=test", "--port=#{port}"
    end

    sleep 1
    system "nc", "-z", "localhost", port.to_s

    assert_match "version: #{version} ()", shell_output("#{bin}/bigquery-emulator --version")
```

File: Formula/b/bigquery-emulator.rb
Comment (carlocab) [inline Formula/b/bigquery-emulator.rb:16]: Let's use `fails_with` to make `brew` use `clang` on Linux instead. You can `grep` for many examples in this repo.

File: Formula/b/bigquery-emulator.rb
Comment (carlocab) [inline Formula/b/bigquery-emulator.rb:11]:
```suggestion
  uses_from_macos "llvm" => :build
```
Since you only need this on Linux.

File: Formula/b/bigquery-emulator.rb
Comment (carlocab) [inline Formula/b/bigquery-emulator.rb:32]: We could maybe pass an extra `-X` flag so we don't have an empty `()` here. (Could maybe make it `(Homebrew)` in place of what I assume would otherwise be a commit hash.)

**Review by carlocab — COMMENTED** [review]
Thanks for the PR, a few comments.

File: Formula/b/bigquery-emulator.rb
Comment (gmpify) [reply Formula/b/bigquery-emulator.rb]: I've pushed this on latest commit, thanks.

File: Formula/b/bigquery-emulator.rb
Comment (gmpify) [reply Formula/b/bigquery-emulator.rb]: Decided to with `(Homebrew)`. Added on latest commit.

File: Formula/b/bigquery-emulator.rb
Comment (carlocab) [inline Formula/b/bigquery-emulator.rb:14]: Does it fail with every GCC version? Do we know the reason why?

Comment (gmpify) [reply Formula/b/bigquery-emulator.rb:14]: idk why, I know it fails with gcc and maintainer [recommends using clang](https://github.com/goccy/bigquery-emulator?tab=readme-ov-file#install).

**Review by carlocab — APPROVED** [review]
Thanks!

**Comment by gmpify** [comment]
Thanks for the review @carlocab!
Is there anything else I should do before this get merged?

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
@gmpify, thanks for your first contribution to homebrew-core!! :tada: 🥇

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!

**Pull request merged**
